crates/noirc_evaluator/src/brillig/brillig_gen/brillig_block.rs

```diff
@@ -497,6 +498,18 @@ impl<'block> BrilligBlock<'block> {
 /// This probably should be explicitly casted in SSA to avoid having to coerce at this level.
 pub(crate) fn type_of_binary_operation(lhs_type: Type, rhs_type: Type) -> Type {
     match (lhs_type, rhs_type) {
+        (_, Type::Function) | (Type::Function, _) => {
+            unreachable!("Functions are invalid in binary operations")
+        }
+        (_, Type::Reference) | (Type::Reference, _) => {
+            unreachable!("References are invalid in binary operations")
```

Comment:
Do we want &1 + &2 to work? I guess not

Reply:
It's not supported in ACIR either so in case we want to implement it should be for both at the same time!

Reply:
> Do we want &1 + &2 to work? I guess not

We do not, `+` is only for integer types currently. If we wanted to implement addition for reference types (among others) we'd need to wait for traits.